Consider enabling `useless-suppression`

### Describe the feature ###
There is a pylint option `useless-suppression` that outputs the lines were a pylint ignore is defined, that didn't cause a pylint warning anyway. It can be enabled with `pylint -e useless-suppression <dirs>`, or by appending it to the `enabled=` parameter in `.pylintrc`.

Not sure if it's worth enabling globally to check during CI, or just run it sometimes and clean up old ignore statements. I also don't know if there are things that Python 2 will warn upon while Python 3 wont (spoiler alert: there probably will be).

This is not for big improvements, but might keep the code a bit cleaner.

For the current master branch, this is the output I have now. I've ran with pylint for python3.

```
************* Module lib.service
resources/lib/service.py:5:0: I0021: Useless suppression of 'non-parent-init-called' (useless-suppression)
resources/lib/service.py:5:0: I0021: Useless suppression of 'no-member' (useless-suppression)
resources/lib/service.py:5:0: I0021: Useless suppression of 'too-many-function-args' (useless-suppression)
************* Module lib.tokenresolver
resources/lib/tokenresolver.py:331:0: I0021: Useless suppression of 'unused-variable' (useless-suppression)
************* Module lib.kodiutils
resources/lib/kodiutils.py:5:0: I0021: Useless suppression of 'too-many-function-args' (useless-suppression)
resources/lib/kodiutils.py:454:0: I0021: Useless suppression of 'unused-variable' (useless-suppression)
************* Module test.test_statichelper
test/test_statichelper.py:5:0: I0021: Useless suppression of 'invalid-name' (useless-suppression)
************* Module test.test_search
test/test_search.py:5:0: I0021: Useless suppression of 'missing-function-docstring' (useless-suppression)
************* Module test.test_language
test/test_language.py:4:0: I0021: Useless suppression of 'missing-function-docstring' (useless-suppression)
************* Module test.xbmc
test/xbmc.py:190:0: I0021: Useless suppression of 'unused-argument' (useless-suppression)
************* Module test.test_settings
test/test_settings.py:4:0: I0021: Useless suppression of 'missing-function-docstring' (useless-suppression)
************* Module test.xbmcvfs
test/xbmcvfs.py:20:0: I0021: Useless suppression of 'too-few-public-methods' (useless-suppression)
************* Module test.test_proxy
test/test_proxy.py:5:0: I0021: Useless suppression of 'duplicate-code' (useless-suppression)
test/test_proxy.py:5:0: I0021: Useless suppression of 'protected-access' (useless-suppression)

--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)
```

--- a/Makefile
+++ b/Makefile
@@ -1,4 +1,5 @@
 ENVS := flake8,py27,py35,py36,py37,py38
+PYTHON := python
 export PYTHONPATH := $(CURDIR)/resources/lib:$(CURDIR)/test
 addon_xml := addon.xml
 
@@ -34,7 +35,11 @@ tox:
 
 pylint:
 	@echo -e "$(white)=$(blue) Starting sanity pylint test$(reset)"
-	pylint resources/lib/ test/
+	$(PYTHON) /usr/bin/pylint resources/lib/ test/
+
+pylint-warnings:
+	@echo -e "$(white)=$(blue) Starting sanity pylint test$(reset)"
+	$(PYTHON) /usr/bin/pylint -e useless-suppression resources/lib/ test/
 
 language:
 	@echo -e "$(white)=$(blue) Checking translations$(reset)"
@@ -42,20 +47,20 @@ language:
 
 addon: clean
 	@echo -e "$(white)=$(blue) Starting sanity addon tests$(reset)"
-	kodi-addon-checker . --branch=krypton
-	kodi-addon-checker . --branch=leia
+	$(PYTHON) /usr/bin/kodi-addon-checker . --branch=krypton
+	$(PYTHON) /usr/bin/kodi-addon-checker . --branch=leia
 
 unit: clean
 	@echo -e "$(white)=$(blue) Starting unit tests$(reset)"
-	-pkill -ef proxy.py
-	proxy.py &
-	python -m unittest discover
-	pkill -ef proxy.py
+	-pkill -ef /usr/bin/proxy.py
+	$(PYTHON) /usr/bin/proxy.py &
+	$(PYTHON) -m unittest discover
+	pkill -ef /usr/bin/proxy.py
 
 run:
 	@echo -e "$(white)=$(blue) Run CLI$(reset)"
-	python resources/lib/service_entry.py
-	python test/run.py /
+	$(PYTHON) resources/lib/service_entry.py
+	$(PYTHON) test/run.py /
 
 zip: clean
 	@echo -e "$(white)=$(blue) Building new package$(reset)"

--- a/resources/lib/kodiutils.py
+++ b/resources/lib/kodiutils.py
@@ -2,8 +2,6 @@
 # GNU General Public License v3.0 (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
 ''' All functionality that requires Kodi imports '''
 
-# pylint: disable=too-many-function-args
-
 from __future__ import absolute_import, division, unicode_literals
 from contextlib import contextmanager
 import xbmc

--- a/resources/lib/service.py
+++ b/resources/lib/service.py
@@ -2,8 +2,6 @@
 # GNU General Public License v3.0 (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
 ''' This is the actual VRT NU service entry point '''
 
-# pylint: disable=non-parent-init-called,no-member,too-many-function-args
-
 from __future__ import absolute_import, division, unicode_literals
 from xbmc import Monitor
 from apihelper import ApiHelper

--- a/resources/lib/tokenresolver.py
+++ b/resources/lib/tokenresolver.py
@@ -324,11 +324,11 @@ class TokenResolver:
         ''' Delete all cached tokens '''
         # Remove old tokens
         # FIXME: Deprecate and simplify this part in a future version
-        dirs, files = listdir(addon_profile())  # pylint: disable=unused-variable
+        _, files = listdir(addon_profile())
         token_files = [item for item in files if item.endswith('.tkn')]
         # Empty userdata/tokens/ directory
         if exists(get_tokens_path()):
-            dirs, files = listdir(get_tokens_path())  # pylint: disable=unused-variable
+            _, files = listdir(get_tokens_path())
             token_files += ['tokens/' + item for item in files]
         if token_files:
             for item in token_files:

--- a/test/test_apihelper.py
+++ b/test/test_apihelper.py
@@ -1,7 +1,8 @@
 # -*- coding: utf-8 -*-
 # GNU General Public License v3.0 (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+''' Unit tests for ApiHelper functionality '''
 
-# pylint: disable=invalid-name,missing-docstring
+# pylint: disable=invalid-name
 
 from __future__ import absolute_import, division, print_function, unicode_literals
 import unittest
@@ -19,12 +20,14 @@ xbmcvfs = __import__('xbmcvfs')
 
 
 class TestApiHelper(unittest.TestCase):
+    ''' TestCase class '''
 
     _favorites = Favorites()
     _resumepoints = ResumePoints()
     _apihelper = ApiHelper(_favorites, _resumepoints)
 
     def test_get_api_data_single_season(self):
+        ''' Test listing episodes for a single season '''
         title_items, sort, ascending, content = self._apihelper.list_episodes(program='het-journaal')
         self.assertTrue(110 <= len(title_items) <= 140, 'We got %s items instead.' % len(title_items))
         self.assertEqual(sort, 'dateadded')
@@ -32,6 +35,7 @@ class TestApiHelper(unittest.TestCase):
         self.assertEqual(content, 'episodes')
 
     def test_get_api_data_multiple_seasons(self):
+        ''' Test listing episodes for multiple seasons '''
         title_items, sort, ascending, content = self._apihelper.list_episodes(program='thuis')
         self.assertTrue(len(title_items) < 5)
         self.assertEqual(sort, 'label')
@@ -39,6 +43,7 @@ class TestApiHelper(unittest.TestCase):
         self.assertEqual(content, 'seasons')
 
     def test_get_api_data_specific_season(self):
+        ''' Test listing episodes for a specific season '''
         title_items, sort, ascending, content = self._apihelper.list_episodes(program='pano')
         self.assertEqual(len(title_items), 5)
         self.assertEqual(sort, 'label')
@@ -46,6 +51,7 @@ class TestApiHelper(unittest.TestCase):
         self.assertEqual(content, 'seasons')
 
     def test_get_api_data_specific_season_without_broadcastdate(self):
+        ''' Test listing episodes without broadcast date '''
         title_items, sort, ascending, content = self._apihelper.list_episodes(program='postbus-x')
         self.assertEqual(len(title_items), 4)
         self.assertEqual(sort, 'label')
@@ -110,12 +116,14 @@ class TestApiHelper(unittest.TestCase):
             self.assertTrue(tvshow.info_dict['studio'] in channel_studios, '%s | %s | %s' % (tvshow.title, tvshow.info_dict['studio'], channel_studios))
 
     def test_get_latest_episode(self):
+        ''' Test getting the latest episode of a program '''
         video = self._apihelper.get_latest_episode(program='het-journaal')
         self.assertTrue(video.get('listitem') is not None)
         self.assertTrue(video.get('video_id') is not None)
         self.assertTrue(video.get('publication_id') is not None)
 
     def test_episode_plot(self):
+        ''' Test getting an episode plot '''
         title_items, sort, ascending, content = self._apihelper.list_episodes(program='thuis', season='allseasons')
         self.assertEqual(sort, 'dateadded')
         self.assertFalse(ascending)
@@ -124,6 +132,7 @@ class TestApiHelper(unittest.TestCase):
         print(kodi_to_ansi(plot))
 
     def test_upnext(self):
+        ''' Test getting next episode '''
         current_episode = dict(
             program='winteruur',
             path='plugin://plugin.video.vrt.nu/play/whatson/705308178527',

--- a/test/test_favorites.py
+++ b/test/test_favorites.py
@@ -1,8 +1,9 @@
 # -*- coding: utf-8 -*-
 # Copyright: (c) 2019, Dag Wieers (@dagwieers) <dag@wieers.com>
 # GNU General Public License v3.0 (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+''' Unit tests for Favorites functionality '''
 
-# pylint: disable=invalid-name,missing-docstring
+# pylint: disable=invalid-name
 
 from __future__ import absolute_import, division, print_function, unicode_literals
 import unittest
@@ -22,6 +23,7 @@ addon.settings['usefavorites'] = 'true'
 
 
 class TestFavorites(unittest.TestCase):
+    ''' TestCase class '''
 
     _favorites = Favorites()
     _resumepoints = ResumePoints()
@@ -45,6 +47,7 @@ class TestFavorites(unittest.TestCase):
 
     @unittest.SkipTest
     def test_unfollow_all(self):
+        ''' Test unfollowing all programs '''
         programs = self._apihelper.get_tvshows()
         for program_item in programs:
             program_title = program_item.get('title')
@@ -56,6 +59,7 @@ class TestFavorites(unittest.TestCase):
 
     @unittest.SkipTest
     def test_follow_number(self):
+        ''' Test following X programs '''
         number = 118
         programs = self._apihelper.get_tvshows()
         shuffle(programs)
@@ -75,6 +79,7 @@ class TestFavorites(unittest.TestCase):
     @unittest.skipUnless(addon.settings.get('username'), 'Skipping as VRT username is missing.')
     @unittest.skipUnless(addon.settings.get('password'), 'Skipping as VRT password is missing.')
     def test_follow_unfollow(self):
+        ''' Test following and unfollowing programs '''
         programs = [
             {'program_title': 'Winteruur', 'program': 'winteruur'},
             {'program_title': 'De Campus Cup', 'program': 'de-campus-cup'},
@@ -94,6 +99,7 @@ class TestFavorites(unittest.TestCase):
             self.assertTrue(self._favorites.is_favorite(program))
 
     def test_programs(self):
+        ''' Test favorite programs list '''
         programs = self._favorites.programs()
         # NOTE: Getting favorites requires credentials
         if addon.settings.get('username') and addon.settings.get('password'):
@@ -101,6 +107,7 @@ class TestFavorites(unittest.TestCase):
         print(programs)
 
     def test_titles(self):
+        ''' Test favorite titles list '''
         titles = self._favorites.titles()
         # NOTE: Getting favorites requires credentials
         if addon.settings.get('username') and addon.settings.get('password'):

--- a/test/test_kodiutils.py
+++ b/test/test_kodiutils.py
@@ -1,7 +1,8 @@
 # -*- coding: utf-8 -*-
 # GNU General Public License v3.0 (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+''' Unit tests for Kodi utilities '''
 
-# pylint: disable=invalid-name,missing-docstring
+# pylint: disable=invalid-name
 
 from __future__ import absolute_import, division, print_function, unicode_literals
 import unittest
@@ -17,11 +18,15 @@ addon = xbmcaddon.Addon()
 
 
 class TestKodiUtils(unittest.TestCase):
+    ''' TestCase class '''
+
     def tearDown(self):
+        ''' Clean up function for TestCase class '''
         xbmc.settings['debug.showloginfo'] = True
         addon.settings['max_log_level'] = '3'
 
     def test_localize(self):
+        ''' Test localizing strings '''
         xbmc.settings['locale.language'] = 'resource.language.nl_nl'
 
         msg = kodiutils.localize(30958)
@@ -38,12 +43,14 @@ class TestKodiUtils(unittest.TestCase):
 
     @staticmethod
     def test_log_disabled():
+        ''' Test with logging disabled '''
         xbmc.settings['debug.showloginfo'] = False
         addon.settings['max_log_level'] = '0'
         kodiutils.log(3, 'No logging when quiet')
 
     @staticmethod
     def test_debug_logging():
+        ''' Test with debug logging enabled '''
         xbmc.settings['debug.showloginfo'] = True
         addon.settings['max_log_level'] = '3'
         kodiutils.log(0, 'Logging as quiet')
@@ -54,19 +61,23 @@ class TestKodiUtils(unittest.TestCase):
 
     @staticmethod
     def test_input_down():
+        ''' Test pressing down key '''
         kodiutils.input_down()
 
     @staticmethod
     def test_container_refresh():
+        ''' Test refreshing the current container '''
         kodiutils.container_refresh()
         kodiutils.container_refresh('plugin://plugin.video.vrt.nu/')
 
     @staticmethod
     def test_container_update():
+        ''' Test updating the current container '''
         kodiutils.container_update()
         kodiutils.container_update('plugin://plugin.video.vrt.nu/')
 
     def test_supports_drm(self):
+        ''' Test if Kodi supports DRM '''
         self.assertTrue(kodiutils.supports_drm())
 
 

--- a/test/test_language.py
+++ b/test/test_language.py
@@ -1,7 +1,8 @@
 # -*- coding: utf-8 -*-
 # GNU General Public License v3.0 (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+''' Unit tests for language-related functionality '''
 
-# pylint: disable=invalid-name,missing-docstring
+# pylint: disable=invalid-name
 
 from __future__ import absolute_import, division, print_function, unicode_literals
 import unittest
@@ -15,7 +16,10 @@ xbmcvfs = __import__('xbmcvfs')
 
 
 class TestLanguage(unittest.TestCase):
+    ''' TestCase class '''
+
     def tearDown(self):
+        ''' Clean up function for TestCase class '''
         xbmc.settings['locale.language'] = 'resource.language.nl_nl'
 
     @staticmethod

--- a/test/test_proxy.py
+++ b/test/test_proxy.py
@@ -1,8 +1,9 @@
 # -*- coding: utf-8 -*-
 # Copyright: (c) 2019, Dag Wieers (@dagwieers) <dag@wieers.com>
 # GNU General Public License v3.0 (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+''' Integration tests for proxy-related functionality '''
 
-# pylint: disable=duplicate-code,invalid-name,missing-docstring,protected-access
+# pylint: disable=invalid-name
 
 from __future__ import absolute_import, division, print_function, unicode_literals
 import sys
@@ -19,41 +20,51 @@ plugin = addon.plugin
 
 @unittest.skipIf(sys.version_info[0] < 3, 'Skipping proxy tests on Python 2')
 class TestProxy(unittest.TestCase):
+    ''' TestCase class '''
+
     def setUp(self):
+        ''' Build up  function for TestCase class '''
         xbmc.settings['network.usehttpproxy'] = True
         xbmc.settings['network.httpproxytype'] = 0
         xbmc.settings['network.httpproxyserver'] = '127.0.0.1'
         xbmc.settings['network.httpproxyport'] = '8899'
 
     def tearDown(self):
+        ''' Clean up function for TestCase class '''
         xbmc.settings['network.usehttpproxy'] = False
 
     # Delete tokens method: '/tokens/delete'
     def test_clear_cookies_route(self):
+        ''' Test clearing cookies '''
         plugin.run(['plugin://plugin.video.vrt.nu/tokens/delete', '0', ''])
         self.assertEqual(plugin.url_for(addon.delete_tokens), 'plugin://plugin.video.vrt.nu/tokens/delete')
 
     # Delete cache method: '/cache/delete'
     def test_invalidate_caches_route(self):
+        ''' Test invalidate caches '''
         plugin.run(['plugin://plugin.video.vrt.nu/cache/delete', '0', ''])
         self.assertEqual(plugin.url_for(addon.delete_cache), 'plugin://plugin.video.vrt.nu/cache/delete')
 
     # A-Z menu: '/programs'
     def test_az_menu(self):
+        ''' Test Programs menu '''
         plugin.run(['plugin://plugin.video.vrt.nu/programs', '0', ''])
         self.assertEqual(plugin.url_for(addon.programs), 'plugin://plugin.video.vrt.nu/programs')
 
     # Categories menu: '/categories'
     def test_categories_menu(self):
+        ''' Test Categories menu '''
         plugin.run(['plugin://plugin.video.vrt.nu/categories', '0', ''])
         self.assertEqual(plugin.url_for(addon.categories), 'plugin://plugin.video.vrt.nu/categories')
 
     # Categories programs menu: '/categories/<category>'
     def test_categories_tvshow_menu(self):
+        ''' Test Categories tvshow '''
         plugin.run(['plugin://plugin.video.vrt.nu/categories/docu', '0', ''])
         self.assertEqual(plugin.url_for(addon.categories, category='docu'), 'plugin://plugin.video.vrt.nu/categories/docu')
 
     # Favorites menu: '/favorites'
     def test_favorites(self):
+        ''' Test Favorites menu '''
         plugin.run(['plugin://plugin.video.vrt.nu/favorites/programs', '0', ''])
         self.assertEqual(plugin.url_for(addon.favorites_programs), 'plugin://plugin.video.vrt.nu/favorites/programs')

--- a/test/test_resumepoints.py
+++ b/test/test_resumepoints.py
@@ -1,8 +1,9 @@
 # -*- coding: utf-8 -*-
 # Copyright: (c) 2019, Dag Wieers (@dagwieers) <dag@wieers.com>
 # GNU General Public License v3.0 (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+''' Unit tests for ResumePoints functionality '''
 
-# pylint: disable=invalid-name,missing-docstring
+# pylint: disable=invalid-name
 
 from __future__ import absolute_import, division, print_function, unicode_literals
 import unittest
@@ -21,6 +22,7 @@ addon.settings['useresumepoints'] = 'true'
 
 
 class TestResumePoints(unittest.TestCase):
+    ''' TestCase class '''
 
     _favorites = Favorites()
     _resumepoints = ResumePoints()
@@ -48,7 +50,8 @@ class TestResumePoints(unittest.TestCase):
 
     @unittest.skipUnless(addon.settings.get('username'), 'Skipping as VRT username is missing.')
     @unittest.skipUnless(addon.settings.get('password'), 'Skipping as VRT password is missing.')
-    def test_update_watchlist(self):
+    def test_update_watchlater(self):
+        ''' Test updating the watch later list '''
         self._resumepoints.refresh(ttl=0)
         asset_id, first_entry = next(iter(list(self._resumepoints._resumepoints.items())))  # pylint: disable=protected-access
         print('%s = %s' % (asset_id, first_entry))
@@ -60,13 +63,15 @@ class TestResumePoints(unittest.TestCase):
         print('%s = %s' % (asset_id, first_entry))
 
     def test_assetpath_to_id(self):
+        ''' Test converting assetpath to assetid '''
         self.assertEqual(None, self._resumepoints.assetpath_to_id(None))
 
         asset_path = '/content/dam/vrt/2019/08/14/woodstock-depot_WP00157456'
         asset_id = 'contentdamvrt20190814woodstockdepotwp00157456'
         self.assertEqual(asset_id, self._resumepoints.assetpath_to_id(asset_path))
 
-    def test_update(self):
+    def test_update_none(self):
+        ''' Test updating empty resumepoints '''
         self.assertTrue(self._resumepoints.update(asset_id=None, title=None, url=None))
 
 

--- a/test/test_routing.py
+++ b/test/test_routing.py
@@ -1,7 +1,8 @@
 # -*- coding: utf-8 -*-
 # GNU General Public License v3.0 (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+''' Integration tests for Routing functionality '''
 
-# pylint: disable=invalid-name,missing-docstring
+# pylint: disable=invalid-name
 
 from __future__ import absolute_import, division, print_function, unicode_literals
 from datetime import datetime, timedelta
@@ -22,13 +23,15 @@ lastweek = now + timedelta(days=-7)
 
 
 class TestRouting(unittest.TestCase):
+    ''' TestCase class '''
 
     def test_main_menu(self):
+        ''' Main menu: / '''
         addon.run(['plugin://plugin.video.vrt.nu/', '0', ''])
         self.assertEqual(plugin.url_for(addon.main_menu), 'plugin://plugin.video.vrt.nu/')
 
-    # Favorites menu: '/favorites'
     def test_favorites(self):
+        ''' Favorites menu: /favorites '''
         addon.run(['plugin://plugin.video.vrt.nu/favorites', '0', ''])
         addon.run(['plugin://plugin.video.vrt.nu/favorites/programs', '0', ''])
         addon.run(['plugin://plugin.video.vrt.nu/favorites/recent', '0', ''])
@@ -39,13 +42,13 @@ class TestRouting(unittest.TestCase):
         addon.run(['plugin://plugin.video.vrt.nu/resumepoints/continue', '0', ''])
         addon.run(['plugin://plugin.video.vrt.nu/favorites/docu', '0', ''])
 
-    # A-Z menu: '/programs'
     def test_az_menu(self):
+        ''' Programs menu: /programs '''
         addon.run(['plugin://plugin.video.vrt.nu/programs', '0', ''])
         self.assertEqual(plugin.url_for(addon.programs), 'plugin://plugin.video.vrt.nu/programs')
 
-    # Episodes menu: '/programs/<program>'
     def test_episodes_menu(self):
+        ''' Episodes menu: /programs/<program> '''
         addon.run(['plugin://plugin.video.vrt.nu/programs/thuis', '0', ''])
         self.assertEqual(plugin.url_for(addon.programs, program='thuis'), 'plugin://plugin.video.vrt.nu/programs/thuis')
         addon.run(['plugin://plugin.video.vrt.nu/programs/pano/2019', '0', ''])
@@ -53,54 +56,54 @@ class TestRouting(unittest.TestCase):
         addon.run(['plugin://plugin.video.vrt.nu/programs/pano/allseasons', '0', ''])
         self.assertEqual(plugin.url_for(addon.programs, program='pano', season='allseasons'), 'plugin://plugin.video.vrt.nu/programs/pano/allseasons')
 
-    # Categories menu: '/categories'
     def test_categories_menu(self):
+        ''' Categories menu: /categories '''
         addon.run(['plugin://plugin.video.vrt.nu/categories', '0', ''])
         self.assertEqual(plugin.url_for(addon.categories), 'plugin://plugin.video.vrt.nu/categories')
 
-    # Categories programs menu: '/categories/<category>'
     def test_categories_tvshow_menu(self):
+        ''' Categories programs menu: /categories/<category> '''
         addon.run(['plugin://plugin.video.vrt.nu/categories/docu', '0', ''])
         self.assertEqual(plugin.url_for(addon.categories, category='docu'), 'plugin://plugin.video.vrt.nu/categories/docu')
         addon.run(['plugin://plugin.video.vrt.nu/categories/voor-kinderen', '0', ''])
         self.assertEqual(plugin.url_for(addon.categories, category='voor-kinderen'), 'plugin://plugin.video.vrt.nu/categories/voor-kinderen')
 
-    # Featured menu: '/featured'
     def test_featured_menu(self):
+        ''' Featured menu: /featured '''
         addon.run(['plugin://plugin.video.vrt.nu/featured', '0', ''])
         self.assertEqual(plugin.url_for(addon.featured), 'plugin://plugin.video.vrt.nu/featured')
 
-    # Featured programs menu: '/featured/<cfeatured>'
     def test_featured_tvshow_menu(self):
+        ''' Featured programs menu: /featured/<cfeatured> '''
         addon.run(['plugin://plugin.video.vrt.nu/featured/kortfilm', '0', ''])
         self.assertEqual(plugin.url_for(addon.featured, feature='kortfilm'), 'plugin://plugin.video.vrt.nu/featured/kortfilm')
 
-    # Channels menu = '/channels/<channel>'
     def test_channels_menu(self):
+        ''' Channels menu = /channels/<channel> '''
         addon.run(['plugin://plugin.video.vrt.nu/channels', '0', ''])
         self.assertEqual(plugin.url_for(addon.channels), 'plugin://plugin.video.vrt.nu/channels')
         addon.run(['plugin://plugin.video.vrt.nu/channels/ketnet', '0', ''])
         self.assertEqual(plugin.url_for(addon.channels, channel='ketnet'), 'plugin://plugin.video.vrt.nu/channels/ketnet')
 
-    # Live TV menu: '/livetv'
     def test_livetv_menu(self):
+        ''' Live TV menu: /livetv '''
         addon.run(['plugin://plugin.video.vrt.nu/livetv', '0', ''])
         self.assertEqual(plugin.url_for(addon.livetv), 'plugin://plugin.video.vrt.nu/livetv')
 
-    # Most recent menu: '/recent/<page>'
     def test_recent_menu(self):
+        ''' Most recent menu: /recent/<page> '''
         addon.run(['plugin://plugin.video.vrt.nu/recent', '0', ''])
         self.assertEqual(plugin.url_for(addon.recent), 'plugin://plugin.video.vrt.nu/recent')
         addon.run(['plugin://plugin.video.vrt.nu/recent/2', '0', ''])
         self.assertEqual(plugin.url_for(addon.recent, page=2), 'plugin://plugin.video.vrt.nu/recent/2')
 
-    # Soon offline menu: '/offline/<page>'
     def test_offline_menu(self):
+        ''' Soon offline menu: /offline/<page> '''
         addon.run(['plugin://plugin.video.vrt.nu/offline', '0', ''])
         self.assertEqual(plugin.url_for(addon.offline), 'plugin://plugin.video.vrt.nu/offline')
 
-    # TV guide menu: '/tvguide/<date>/<channel>'
     def test_tvguide_date_menu(self):
+        ''' TV guide menu: /tvguide/<date>/<channel> '''
         addon.run(['plugin://plugin.video.vrt.nu/tvguide', '0', ''])
         self.assertEqual(plugin.url_for(addon.tvguide), 'plugin://plugin.video.vrt.nu/tvguide/date')
         addon.run(['plugin://plugin.video.vrt.nu/tvguide/date/today', '0', ''])
@@ -112,10 +115,10 @@ class TestRouting(unittest.TestCase):
         addon.run(['plugin://plugin.video.vrt.nu/tvguide/channel/canvas/today', '0', ''])
         self.assertEqual(plugin.url_for(addon.tvguide_channel, channel='canvas', date='today'), 'plugin://plugin.video.vrt.nu/tvguide/channel/canvas/today')
 
-    # Clear search history: '/search/clear'
-    # Add search keyword: '/search/add/<keywords>'
-    # Remove search keyword: '/search/remove/<keywords>'
     def test_search_history(self):
+        ''' Add search keyword: /search/add/<keywords>
+            Clear search history: /search/clear
+            Remove search keyword: /search/remove/<keywords> '''
         addon.run(['plugin://plugin.video.vrt.nu/search/add/foobar', '0', ''])
         self.assertEqual(plugin.url_for(addon.add_search, keywords='foobar'), 'plugin://plugin.video.vrt.nu/search/add/foobar')
         addon.run(['plugin://plugin.video.vrt.nu/search/add/foobar', '0', ''])
@@ -133,8 +136,8 @@ class TestRouting(unittest.TestCase):
         addon.run(['plugin://plugin.video.vrt.nu/search/add/foobar', '0', ''])
         self.assertEqual(plugin.url_for(addon.add_search, keywords='foobar'), 'plugin://plugin.video.vrt.nu/search/add/foobar')
 
-    # Search VRT NU menu: '/search/query/<keywords>/<page>'
     def test_search_menu(self):
+        ''' Search VRT NU menu: /search/query/<keywords>/<page> '''
         addon.run(['plugin://plugin.video.vrt.nu/search', '0', ''])
         self.assertEqual(plugin.url_for(addon.search), 'plugin://plugin.video.vrt.nu/search')
         addon.run(['plugin://plugin.video.vrt.nu/search/query', '0', ''])
@@ -146,84 +149,84 @@ class TestRouting(unittest.TestCase):
         addon.run(['plugin://plugin.video.vrt.nu/search/query/winter', '0', ''])
         self.assertEqual(plugin.url_for(addon.search_query, keywords='winter'), 'plugin://plugin.video.vrt.nu/search/query/winter')
 
-    # Follow method: '/follow/<program>/<title>'
     def test_follow_route(self):
+        ''' Follow method: /follow/<program>/<title> '''
         addon.run(['plugin://plugin.video.vrt.nu/follow/thuis/Thuis', '0', ''])
         self.assertEqual(plugin.url_for(addon.follow, program='thuis', title='Thuis'), 'plugin://plugin.video.vrt.nu/follow/thuis/Thuis')
 
-    # Unfollow method: '/unfollow/<program>/<title>'
     def test_unfollow_route(self):
+        ''' Unfollow method: /unfollow/<program>/<title> '''
         addon.run(['plugin://plugin.video.vrt.nu/unfollow/thuis/Thuis', '0', ''])
         self.assertEqual(plugin.url_for(addon.unfollow, program='thuis', title='Thuis'), 'plugin://plugin.video.vrt.nu/unfollow/thuis/Thuis')
 
-    # Delete tokens method: '/tokens/delete'
     def test_clear_cookies_route(self):
+        ''' Delete tokens method: /tokens/delete '''
         addon.run(['plugin://plugin.video.vrt.nu/tokens/delete', '0', ''])
         self.assertEqual(plugin.url_for(addon.delete_tokens), 'plugin://plugin.video.vrt.nu/tokens/delete')
 
-    # Delete cache method: '/cache/delete'
     def test_invalidate_caches_route(self):
+        ''' Delete cache method: /cache/delete '''
         addon.run(['plugin://plugin.video.vrt.nu/cache/delete', '0', ''])
         self.assertEqual(plugin.url_for(addon.delete_cache), 'plugin://plugin.video.vrt.nu/cache/delete')
 
-    # Refresh favorites method: '/favorites/refresh'
     def test_refresh_favorites_route(self):
+        ''' Refresh favorites method: /favorites/refresh '''
         addon.run(['plugin://plugin.video.vrt.nu/favorites/refresh', '0', ''])
         self.assertEqual(plugin.url_for(addon.favorites_refresh), 'plugin://plugin.video.vrt.nu/favorites/refresh')
 
-    # Refresh resumepoints method: '/resumepoints/refresh'
     def test_refresh_resumepoints_route(self):
+        ''' Refresh resumepoints method: /resumepoints/refresh '''
         addon.run(['plugin://plugin.video.vrt.nu/resumepoints/refresh', '0', ''])
         self.assertEqual(plugin.url_for(addon.resumepoints_refresh), 'plugin://plugin.video.vrt.nu/resumepoints/refresh')
 
-    # Manage favorites method: '/favorites/manage'
     def test_manage_favorites_route(self):
+        ''' Manage favorites method: /favorites/manage '''
         addon.run(['plugin://plugin.video.vrt.nu/favorites/manage', '0', ''])
         self.assertEqual(plugin.url_for(addon.favorites_manage), 'plugin://plugin.video.vrt.nu/favorites/manage')
 
-    # Watch later method: 'plugin://plugin.video.vrt.nu/watchlater/<url>/<asset_id>/<title>'
     def test_watchlater_route(self):
+        ''' Watch later method: plugin://plugin.video.vrt.nu/watchlater/<url>/<asset_id>/<title> '''
         addon.run(['plugin://plugin.video.vrt.nu/watchlater//vrtnu/a-z/winteruur/5/winteruur-s5a1//contentdamvrt20191015winteruurr005a0001depotwp00162177/Winteruur', '0', ''])
         self.assertEqual(plugin.url_for(addon.watchlater, url='/vrtnu/a-z/winteruur/5/winteruur-s5a1', asset_id='/contentdamvrt20191015winteruurr005a0001depotwp00162177', title='Winteruur'), 'plugin://plugin.video.vrt.nu/watchlater//vrtnu/a-z/winteruur/5/winteruur-s5a1//contentdamvrt20191015winteruurr005a0001depotwp00162177/Winteruur')
 
-    # Unwatch later method: 'plugin://plugin.video.vrt.nu/unwatchlater/<url>/<asset_id>/<title>'
     def test_unwatchlater_route(self):
+        ''' Unwatch later method: plugin://plugin.video.vrt.nu/unwatchlater/<url>/<asset_id>/<title> '''
         addon.run(['plugin://plugin.video.vrt.nu/unwatchlater//vrtnu/a-z/winteruur/5/winteruur-s5a1//contentdamvrt20191015winteruurr005a0001depotwp00162177/Winteruur', '0', ''])
         self.assertEqual(plugin.url_for(addon.unwatchlater, url='/vrtnu/a-z/winteruur/5/winteruur-s5a1', asset_id='/contentdamvrt20191015winteruurr005a0001depotwp00162177', title='Winteruur'), 'plugin://plugin.video.vrt.nu/unwatchlater//vrtnu/a-z/winteruur/5/winteruur-s5a1//contentdamvrt20191015winteruurr005a0001depotwp00162177/Winteruur')
 
-    # Play on demand by id = '/play/id/<publication_id>/<video_id>'
-    # Achterflap episode 8 available until 31/12/2025
     def test_play_on_demand_by_id_route(self):
+        ''' Play on demand by id: /play/id/<publication_id>/<video_id> '''
+        # Achterflap episode 8 available until 31/12/2025
         addon.run(['plugin://plugin.video.vrt.nu/play/id/vid-f80fa527-6759-45a7-908d-ec6f0a7b164e/pbs-pub-1a170972-dea3-4ea3-8c27-62d2442ee8a3', '0', ''])
         self.assertEqual(plugin.url_for(addon.play_id,
                                         video_id='vid-f80fa527-6759-45a7-908d-ec6f0a7b164e',
                                         publication_id='pbs-pub-1a170972-dea3-4ea3-8c27-62d2442ee8a3'),
                          'plugin://plugin.video.vrt.nu/play/id/vid-f80fa527-6759-45a7-908d-ec6f0a7b164e/pbs-pub-1a170972-dea3-4ea3-8c27-62d2442ee8a3')
 
-    # Play livestream by id = '/play/id/<video_id>'
-    # Canvas livestream
     def test_play_livestream_by_id_route(self):
+        ''' Play livestream by id: /play/id/<video_id> '''
+        # Canvas livestream
         addon.run(['plugin://plugin.video.vrt.nu/play/id/vualto_canvas_geo', '0', ''])
         self.assertEqual(plugin.url_for(addon.play_id, video_id='vualto_canvas_geo'), 'plugin://plugin.video.vrt.nu/play/id/vualto_canvas_geo')
 
-    # Play on demand by url = '/play/url/<vrtnuwebsite_url>'
-    # Achterflap episode 8 available until 31/12/2025
     def test_play_on_demand_by_url_route(self):
+        ''' Play on demand by url: /play/url/<vrtnuwebsite_url> '''
+        # Achterflap episode 8 available until 31/12/2025
         addon.run(['plugin://plugin.video.vrt.nu/play/url/https://www.vrt.be/vrtnu/a-z/achterflap/1/achterflap-s1a8/', '0', ''])
         self.assertEqual(plugin.url_for(addon.play_url,
                                         video_url='https://www.vrt.be/vrtnu/a-z/achterflap/1/achterflap-s1a8/'),
                          'plugin://plugin.video.vrt.nu/play/url/https://www.vrt.be/vrtnu/a-z/achterflap/1/achterflap-s1a8/')
 
-    # Play livestream by url = '/play/url/<vrtnuwebsite_url>'
-    # Canvas livestream
     def test_play_livestream_by_url_route(self):
+        ''' Play livestream by url: /play/url/<vrtnuwebsite_url> '''
+        # Canvas livestream
         addon.run(['plugin://plugin.video.vrt.nu/play/url/https://www.vrt.be/vrtnu/kanalen/canvas/', '0', ''])
         self.assertEqual(plugin.url_for(addon.play_url,
                                         video_url='https://www.vrt.be/vrtnu/kanalen/canvas/'),
                          'plugin://plugin.video.vrt.nu/play/url/https://www.vrt.be/vrtnu/kanalen/canvas/')
 
-    # Play last episode method = '/play/lastepisode/<program>'
     def test_play_latestepisode_route(self):
+        ''' Play last episode method: /play/lastepisode/<program> '''
         addon.run(['plugin://plugin.video.vrt.nu/play/latest/het-journaal', '0', ''])
         self.assertEqual(plugin.url_for(addon.play_latest, program='het-journaal'), 'plugin://plugin.video.vrt.nu/play/latest/het-journaal')
         addon.run(['plugin://plugin.video.vrt.nu/play/latest/terzake', '0', ''])
@@ -231,8 +234,8 @@ class TestRouting(unittest.TestCase):
         addon.run(['plugin://plugin.video.vrt.nu/play/latest/winteruur', '0', ''])
         self.assertEqual(plugin.url_for(addon.play_latest, program='winteruur'), 'plugin://plugin.video.vrt.nu/play/latest/winteruur')
 
-    # Play episode by air date method = '/play/airdate/<channel>/<start_date>'
     def test_play_airdateepisode_route(self):
+        ''' Play episode by air date method: /play/airdate/<channel>/<start_date> '''
         # Test Het Journaal
         addon.run([lastweek.strftime('plugin://plugin.video.vrt.nu/play/airdate/een/%Y-%m-%dT19:00:00'), '0', ''])
         self.assertEqual(plugin.url_for(addon.play_by_air_date,
@@ -246,8 +249,8 @@ class TestRouting(unittest.TestCase):
                                         start_date=lastweek.strftime('%Y-%m-%dT20:00:00')),
                          lastweek.strftime('plugin://plugin.video.vrt.nu/play/airdate/canvas/%Y-%m-%dT20:00:00'))
 
-    # Play Up Next episode method = '/play/upnext/<video_id>'
     def test_play_upnext_route(self):
+        ''' Play Up Next episode method: /play/upnext/<video_id> '''
         addon.run(['plugin://plugin.video.vrt.nu/play/upnext/vid-a39ab219-9598-4a79-b676-98b724cceff1', '0', ''])
         self.assertEqual(plugin.url_for(addon.play_upnext, video_id='vid-a39ab219-9598-4a79-b676-98b724cceff1'), 'plugin://plugin.video.vrt.nu/play/upnext/vid-a39ab219-9598-4a79-b676-98b724cceff1')
 

--- a/test/test_search.py
+++ b/test/test_search.py
@@ -1,8 +1,9 @@
 # -*- coding: utf-8 -*-
 # Copyright: (c) 2019, Dag Wieers (@dagwieers) <dag@wieers.com>
 # GNU General Public License v3.0 (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+''' Integration tests for search functionality '''
 
-# pylint: disable=invalid-name,missing-docstring
+# pylint: disable=invalid-name
 
 from __future__ import absolute_import, division, print_function, unicode_literals
 import unittest
@@ -19,6 +20,7 @@ xbmcvfs = __import__('xbmcvfs')
 
 
 class TestSearch(unittest.TestCase):
+    ''' TestClass class '''
 
     _favorites = Favorites()
     _resumepoints = ResumePoints()

--- a/test/test_settings.py
+++ b/test/test_settings.py
@@ -1,7 +1,8 @@
 # -*- coding: utf-8 -*-
 # GNU General Public License v3.0 (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+''' Integration tests for changes to settings '''
 
-# pylint: disable=invalid-name,missing-docstring
+# pylint: disable=invalid-name
 
 from __future__ import absolute_import, division, print_function, unicode_literals
 import unittest
@@ -18,7 +19,10 @@ addon = xbmcaddon.Addon()
 
 
 class TestSettings(unittest.TestCase):
+    ''' TestCase class '''
+
     def tearDown(self):
+        ''' Clean up function for TestCase class '''
         addon.settings['usemenucaching'] = 'true'
         addon.settings['usehttpcaching'] = 'true'
         addon.settings['usefavorites'] = 'true'

--- a/test/test_statichelper.py
+++ b/test/test_statichelper.py
@@ -1,8 +1,7 @@
 # -*- coding: utf-8 -*-
 # Copyright: (c) 2019, Dag Wieers (@dagwieers) <dag@wieers.com>
 # GNU General Public License v3.0 (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
-
-# pylint: disable=invalid-name,missing-docstring
+''' Tests for StaticHelper functionality '''
 
 from __future__ import absolute_import, division, print_function, unicode_literals
 import unittest
@@ -10,8 +9,10 @@ import statichelper
 
 
 class TestStaticHelper(unittest.TestCase):
+    ''' TestCase class '''
 
     def test_url_to_episode(self):
+        ''' Test converting URL to episode '''
         long_url = 'https://www.vrt.be/vrtnu/a-z/buck/1/buck-s1a32/'
         episode = '/vrtnu/a-z/buck/1/buck-s1a32/'
         self.assertEqual(episode, statichelper.url_to_episode(long_url))
@@ -23,23 +24,20 @@ class TestStaticHelper(unittest.TestCase):
         self.assertEqual(None, statichelper.url_to_episode('foobar'))
 
     def test_url_to_program(self):
-        long_url = 'https://www.vrt.be/vrtnu/a-z/buck/1/buck-s1a32/'
+        ''' Test converting URL to program '''
         program = 'buck'
-        self.assertEqual(program, statichelper.url_to_program(long_url))
-
-        medium_url = '//www.vrt.be/vrtnu/a-z/buck/1/buck-s1a32/'
-        program = 'buck'
-        self.assertEqual(program, statichelper.url_to_program(medium_url))
-
         short_url = '/vrtnu/a-z/buck/1/buck-s1a32/'
-        program = 'buck'
-        self.assertEqual(program, statichelper.url_to_program(short_url))
+        medium_url = '//www.vrt.be/vrtnu/a-z/buck/1/buck-s1a32/'
+        long_url = 'https://www.vrt.be/vrtnu/a-z/buck/1/buck-s1a32/'
+        short_relevant_url = '/vrtnu/a-z/buck.relevant/'
 
-        short_url = '/vrtnu/a-z/buck.relevant/'
-        program = 'buck'
+        self.assertEqual(program, statichelper.url_to_program(long_url))
+        self.assertEqual(program, statichelper.url_to_program(medium_url))
         self.assertEqual(program, statichelper.url_to_program(short_url))
+        self.assertEqual(program, statichelper.url_to_program(short_relevant_url))
 
     def test_program_to_url(self):
+        ''' Test converting program to URL '''
         program = 'de-campus-cup'
         short_url = '/vrtnu/a-z/de-campus-cup/'
         medium_url = '//www.vrt.be/vrtnu/a-z/de-campus-cup/'
@@ -50,6 +48,7 @@ class TestStaticHelper(unittest.TestCase):
         self.assertEqual(long_url, statichelper.program_to_url(program, 'long'))
 
     def test_video_to_api_url(self):
+        ''' Test convert video to api URL '''
         video = 'https://www.vrt.be/vrtnu/a-z/de-ideale-wereld/2019-nj/de-ideale-wereld-d20191010/'
         api_url = '//www.vrt.be/vrtnu/a-z/de-ideale-wereld/2019-nj/de-ideale-wereld-d20191010/'
         self.assertEqual(api_url, statichelper.video_to_api_url(video))
@@ -59,6 +58,7 @@ class TestStaticHelper(unittest.TestCase):
         self.assertEqual(api_url, statichelper.video_to_api_url(video))
 
     def test_play_url_to_id(self):
+        ''' Test converting play_url to play_id '''
         url = 'plugin://plugin.video.vrt.nu/play/id/vid-5b12c0f6-b8fe-426f-a600-557f501f3be9/pbs-pub-7e2764cf-a8c0-4e78-9cbc-46d39381c237'
         play_id = dict(video_id='vid-5b12c0f6-b8fe-426f-a600-557f501f3be9')
         self.assertEqual(play_id, statichelper.play_url_to_id(url))
@@ -72,6 +72,7 @@ class TestStaticHelper(unittest.TestCase):
         self.assertEqual(play_id, statichelper.play_url_to_id(url))
 
     def test_reformat_url(self):
+        ''' Test reformatting URLs '''
         short_url = '/vrtnu/a-z/terzake/2019/terzake-d20191017/'
         medium_url = '//www.vrt.be/vrtnu/a-z/terzake/2019/terzake-d20191017/'
         long_url = 'https://www.vrt.be/vrtnu/a-z/terzake/2019/terzake-d20191017/'
@@ -93,15 +94,17 @@ class TestStaticHelper(unittest.TestCase):
         self.assertEqual(short_url, statichelper.reformat_url(long_url + '#foo', 'short'))
 
     def test_shorten_link(self):
+        ''' Test shortening links '''
+        permalink = 'vrtnu.be/p.LR90GkqOD'
+        medium_url = '//www.vrt.be/vrtnu/p.LR90GkqOD'
+        long_url = 'https://www.vrt.be/vrtnu/p.LR90GkqOD'
+
+        self.assertEqual(permalink, statichelper.shorten_link(long_url))
+        self.assertEqual(permalink, statichelper.shorten_link(medium_url))
         self.assertEqual(None, statichelper.shorten_link(None))
 
-        long_url = 'https://www.vrt.be/vrtnu/p.LR90GkqOD'
-        self.assertEqual('vrtnu.be/p.LR90GkqOD', statichelper.shorten_link(long_url))
-
-        medium_url = '//www.vrt.be/vrtnu/p.LR90GkqOD'
-        self.assertEqual('vrtnu.be/p.LR90GkqOD', statichelper.shorten_link(medium_url))
-
     def test_realpage(self):
+        ''' Test converting input to page '''
         self.assertEqual(1, statichelper.realpage('foo'))
         self.assertEqual(1, statichelper.realpage('-1'))
         self.assertEqual(1, statichelper.realpage('0'))
@@ -109,6 +112,7 @@ class TestStaticHelper(unittest.TestCase):
         self.assertEqual(3, statichelper.realpage('3'))
 
     def test_capitalize(self):
+        ''' Test capitalizing string '''
         self.assertEqual('Foo bar', statichelper.capitalize('foo bar'))
         self.assertEqual('Foo bar', statichelper.capitalize('Foo bar'))
         self.assertEqual('FoO bAr', statichelper.capitalize('foO bAr'))

--- a/test/test_streamservice.py
+++ b/test/test_streamservice.py
@@ -1,7 +1,8 @@
 # -*- coding: utf-8 -*-
 # GNU General Public License v3.0 (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+''' Unit tests for StreamService functionality '''
 
-# pylint: disable=invalid-name,missing-docstring
+# pylint: disable=invalid-name
 
 from __future__ import absolute_import, division, print_function, unicode_literals
 from datetime import datetime, timedelta
@@ -32,6 +33,7 @@ yesterday = now + timedelta(days=-1)
 
 
 class TestStreamService(unittest.TestCase):
+    ''' TestCase class '''
 
     _tokenresolver = TokenResolver()
     _streamservice = StreamService(_tokenresolver)
@@ -39,6 +41,7 @@ class TestStreamService(unittest.TestCase):
     @unittest.skipUnless(addon.settings.get('username'), 'Skipping as VRT username is missing.')
     @unittest.skipUnless(addon.settings.get('password'), 'Skipping as VRT password is missing.')
     def test_get_ondemand_stream_from_invalid_url(self):
+        ''' Test getting stream from invalid URL '''
         video = dict(video_url='https://www.vrt.be/vrtnu/a-z/het-journaal/2017/het-journaal-het-journaal-laat-20170501/', video_id=None, publication_id=None)
         try:
             stream = self._streamservice.get_stream(video)
@@ -49,6 +52,7 @@ class TestStreamService(unittest.TestCase):
     @unittest.skipUnless(addon.settings.get('username'), 'Skipping as VRT username is missing.')
     @unittest.skipUnless(addon.settings.get('password'), 'Skipping as VRT password is missing.')
     def test_get_ondemand_stream_from_url_gets_stream_does_not_crash(self):
+        ''' Test getting stream from URL does not crash '''
         video = dict(video_url=yesterday.strftime('https://www.vrt.be/vrtnu/a-z/het-journaal/2019/het-journaal-het-journaal-laat-%Y%m%d/'),
                      video_id=None,
                      publication_id=None)
@@ -56,6 +60,7 @@ class TestStreamService(unittest.TestCase):
         self.assertTrue(stream is not None)
 
     def test_get_mpd_live_stream_from_url_does_not_crash_returns_stream_and_licensekey(self):
+        ''' Test getting MPD stream from URL '''
         addon.settings['usedrm'] = 'true'
         addon.settings['useinputstreamadaptive'] = 'true'
         video = dict(video_url=CHANNELS[1]['live_stream'], video_id=None, publication_id=None)
@@ -66,6 +71,7 @@ class TestStreamService(unittest.TestCase):
 #            self.assertTrue(stream.license_key is not None)
 
     def test_get_hls_live_stream_from_url_does_not_crash_returns_stream_and_licensekey(self):
+        ''' Test getting HLS stream from URL '''
         addon.settings['usedrm'] = 'false'
         addon.settings['useinputstreamadaptive'] = 'false'
         video = dict(video_url=CHANNELS[1]['live_stream'], video_id=None, publication_id=None)
@@ -75,6 +81,7 @@ class TestStreamService(unittest.TestCase):
             self.assertTrue(stream is not None)
 
     def test_get_mpd_live_stream_from_url_does_not_crash(self):
+        ''' Test getting MPD stream from URL '''
         addon.settings['usedrm'] = 'false'
         video = dict(video_url=CHANNELS[1]['live_stream'], video_id=None, publication_id=None)
         stream = self._streamservice.get_stream(video)
@@ -83,6 +90,7 @@ class TestStreamService(unittest.TestCase):
             self.assertTrue(stream is not None)
 
     def test_get_hls_live_stream_from_url_does_not_crash(self):
+        ''' Test getting HLS stream from URL '''
         addon.settings['usedrm'] = 'false'
         addon.settings['useinputstreamadaptive'] = 'false'
         video = dict(video_url=CHANNELS[1]['live_stream'], video_id=None, publication_id=None)
@@ -92,6 +100,7 @@ class TestStreamService(unittest.TestCase):
             self.assertTrue(stream is not None)
 
     def test_get_mpd_live_stream_from_id_does_not_crash(self):
+        ''' Test getting MPD stream from URL '''
         addon.settings['usedrm'] = 'false'
         addon.settings['useinputstreamadaptive'] = 'true'
         video = dict(video_url=None, video_id=CHANNELS[1]['live_stream_id'], publication_id=None)
@@ -107,6 +116,7 @@ class TestStreamService(unittest.TestCase):
             self.assertTrue(stream is not None)
 
     def test_get_hls_live_stream_from_id_does_not_crash(self):
+        ''' Test getting HLS stream from URL '''
         addon.settings['usedrm'] = 'false'
         addon.settings['useinputstreamadaptive'] = 'false'
         video = dict(video_url=None, video_id=CHANNELS[1]['live_stream_id'], publication_id=None)

--- a/test/test_tokenresolver.py
+++ b/test/test_tokenresolver.py
@@ -1,7 +1,8 @@
 # -*- coding: utf-8 -*-
 # GNU General Public License v3.0 (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+''' Unit tests for TokenResolver functionality '''
 
-# pylint: disable=invalid-name,missing-docstring
+# pylint: disable=invalid-name
 
 from __future__ import absolute_import, division, print_function, unicode_literals
 import unittest
@@ -17,43 +18,53 @@ addon = xbmcaddon.Addon()
 
 
 class TestTokenResolver(unittest.TestCase):
+    ''' TestCase class '''
+
     _tokenresolver = TokenResolver()
     username = None
     password = None
 
     def setUp(self):
+        ''' Build up  function for TestCase class '''
         # Save password
         self.username = addon.settings['username']
         self.password = addon.settings['password']
 
     def tearDown(self):
+        ''' Clean up function for TestCase class '''
         # Restore password
         addon.settings['username'] = self.username
         addon.settings['password'] = self.password
 
     def test_refresh_login(self):
+        ''' Test refreshing login '''
         self._tokenresolver.refresh_login()
 
     def test_cleanup_userdata(self):
+        ''' Test cleaning up userdata '''
         self._tokenresolver.cleanup_userdata()
 
-    def test_successful(self):
+    def test_successful_login(self):
+        ''' Test successful login '''
         self.username = addon.settings['username']
         self.password = addon.settings['password']
         self._tokenresolver.login(refresh=False)
 
     def test_invalid_login(self):
+        ''' Test invalid login '''
         addon.settings['username'] = 'foo'
         addon.settings['password'] = 'bar'
         self._tokenresolver.login(refresh=False)
 
     def test_missing_username(self):
+        ''' Test missing username '''
         addon.settings['username'] = ''
         addon.settings['password'] = self.password
         self._tokenresolver.login(refresh=True)
         self._tokenresolver.login(refresh=False)
 
     def test_missing_password(self):
+        ''' Test missing password '''
         addon.settings['username'] = self.username
         addon.settings['password'] = ''
         self._tokenresolver.login(refresh=True)

--- a/test/test_tvguide.py
+++ b/test/test_tvguide.py
@@ -1,8 +1,9 @@
 # -*- coding: utf-8 -*-
 # Copyright: (c) 2019, Dag Wieers (@dagwieers) <dag@wieers.com>
 # GNU General Public License v3.0 (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+''' Unit tests for TVGuide functionality '''
 
-# pylint: disable=invalid-name,missing-docstring
+# pylint: disable=invalid-name
 
 from __future__ import absolute_import, division, print_function, unicode_literals
 from datetime import datetime, timedelta
@@ -23,6 +24,7 @@ channels = ['een', 'canvas', 'ketnet']
 
 
 class TestTVGuide(unittest.TestCase):
+    ''' TestCase class '''
 
     _tvguide = TVGuide()
 
@@ -65,6 +67,7 @@ class TestTVGuide(unittest.TestCase):
         self.assertEqual(episode_items, [])
 
     def test_livetv_description(self):
+        ''' Test Live TV description '''
         description = self._tvguide.live_description('een')
         print(kodi_to_ansi(description))
         description = self._tvguide.live_description('canvas')
@@ -82,6 +85,7 @@ class TestTVGuide(unittest.TestCase):
         self.assertTrue(episode_items)
 
     def test_parse(self):
+        ''' Test parsing date '''
         now = datetime.now(dateutil.tz.tzlocal())
         date = self._tvguide.parse('2019-05-11', now)
         print(date)

--- a/test/test_vrtplayer.py
+++ b/test/test_vrtplayer.py
@@ -1,7 +1,8 @@
 # -*- coding: utf-8 -*-
 # GNU General Public License v3.0 (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+''' Unit tests for VRTPlayer functionality '''
 
-# pylint: disable=invalid-name,missing-docstring
+# pylint: disable=invalid-name
 
 from __future__ import absolute_import, division, print_function, unicode_literals
 import random
@@ -20,6 +21,7 @@ xbmcvfs = __import__('xbmcvfs')
 
 
 class TestVRTPlayer(unittest.TestCase):
+    ''' TestCase class '''
 
     _favorites = Favorites()
     _resumepoints = ResumePoints()
@@ -27,6 +29,7 @@ class TestVRTPlayer(unittest.TestCase):
     _vrtplayer = VRTPlayer()
 
     def test_show_videos_single_episode_shows_videos(self):
+        ''' Test listing single episode for a given program '''
         program = 'marathonradio'
         episode_items, sort, ascending, content = self._apihelper.list_episodes(program=program)
         self.assertTrue(episode_items, msg=program)
@@ -37,6 +40,7 @@ class TestVRTPlayer(unittest.TestCase):
         self._vrtplayer.show_episodes_menu(program)
 
     def test_show_videos_single_season_shows_videos(self):
+        ''' Test listing single season for a given program '''
         program = 'het-weer'
         episode_items, sort, ascending, content = self._apihelper.list_episodes(program=program)
         self.assertTrue(episode_items, msg=program)
@@ -47,6 +51,7 @@ class TestVRTPlayer(unittest.TestCase):
         self._vrtplayer.show_episodes_menu(program)
 
     def test_show_videos_multiple_seasons_shows_videos(self):
+        ''' Test listing multiple season for a given program '''
         program = 'pano'
         episode_items, sort, ascending, content = self._apihelper.list_episodes(program=program)
         self.assertTrue(episode_items)
@@ -57,6 +62,7 @@ class TestVRTPlayer(unittest.TestCase):
         self._vrtplayer.show_episodes_menu(program)
 
     def test_show_videos_specific_seasons_shows_videos(self):
+        ''' Test listing specific seasons for a given program '''
         program = 'thuis'
         episode_items, sort, ascending, content = self._apihelper.list_episodes(program=program)
         self.assertTrue(episode_items, msg=program)

--- a/test/xbmc.py
+++ b/test/xbmc.py
@@ -187,7 +187,7 @@ def executeJSONRPC(jsonrpccommand):
     return json.dumps(dict(error=dict(code=-1, message='Not implemented'), id=1, jsonrpc='2.0'))
 
 
-def getCondVisibility(string):  # pylint: disable=unused-argument
+def getCondVisibility(string):
     ''' A reimplementation of the xbmc getCondVisibility() function '''
     if string == 'system.platform.android':
         return False

--- a/test/xbmcvfs.py
+++ b/test/xbmcvfs.py
@@ -17,7 +17,7 @@ def File(path, flags='r'):
 def Stat(path):
     ''' A reimplementation of the xbmcvfs Stat() function '''
 
-    class stat:  # pylint: disable=too-few-public-methods
+    class stat:
         ''' A reimplementation of the xbmcvfs stat class '''
 
         def __init__(self, path):

